### PR TITLE
feat: remove portal commands from apic-extension

### PR DESCRIPTION
**Pull request by blackchoey**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

az apic service portal *

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/apic-extension/azext_apic_extension/aaz/latest/apic/service/_import_from_apim.py
Comment (blackchoey) [inline src/apic-extension/azext_apic_extension/aaz/latest/apic/service/_import_from_apim.py:25]: @arpishahmsft the rest api spec 2024-03-01 does not contain import from apim definition, should I use `2024-03-15-preview` here, or manually change back to `2024-03-01`?

**Comment by blackchoey** [comment]
@jsntcy this is my PR to contribute Azure CLI extension, please hold on PR merge until I get approval from my teammate.
I also have some questions regarding the auto generated code. It would be great if you could help answer some of them.
Thank you!

File: src/apic-extension/azext_apic_extension/azext_metadata.json
Comment (blackchoey) [inline src/apic-extension/azext_apic_extension/azext_metadata.json:3]: I don't know why the min CLI core version was bumped here. Am I able to keep the min version unchanged? So the new version will not require users upgrade their Azure CLI.

File: src/apic-extension/azext_apic_extension/aaz/latest/__init__.py
Comment (blackchoey) [inline src/apic-extension/azext_apic_extension/aaz/latest/__init__.py:8]: Minor: the auto generated changes looks good to me, but I don't know why it's not included in previous version. Can anyone share some backgrounds?

File: src/apic-extension/azext_apic_extension/aaz/latest/apic/service/_import_from_apim.py
Comment (blackchoey) [reply src/apic-extension/azext_apic_extension/aaz/latest/apic/service/_import_from_apim.py]: @arpishahmsft Manually changed to `2024-03-01` per your suggestion

File: src/apic-extension/azext_apic_extension/aaz/latest/__init__.py
Comment (AllyW) [reply src/apic-extension/azext_apic_extension/aaz/latest/__init__.py:8]: The style-suppress default code line is added into aaz-dev tool in June 2022.

File: src/apic-extension/azext_apic_extension/azext_metadata.json
Comment (AllyW) [reply src/apic-extension/azext_apic_extension/azext_metadata.json:3]: AAZ-dev tool might generate code that requires newly introduced functionalities in azure cli recent versions. 
If you reviewed the generated code and did not find any new features in aaz code, then you can keep the previous cli min version. Please run live tests and confirm its consistency.

Comment (blackchoey) [reply src/apic-extension/azext_apic_extension/azext_metadata.json:3]: Thank you!

File: src/apic-extension/azext_apic_extension/aaz/latest/apic/service/_create.py
Comment (arpishahmsft) [inline src/apic-extension/azext_apic_extension/aaz/latest/apic/service/_create.py:287]: I believe we still need this in all of the service commands

Comment (blackchoey) [reply src/apic-extension/azext_apic_extension/aaz/latest/apic/service/_create.py:287]: Searched the code and didn't find usage of `data_api_hostname`, can you help point out where it's used?

Comment (blackchoey) [reply src/apic-extension/azext_apic_extension/aaz/latest/apic/service/_create.py]: Added the properties back

**Review by AllyW — APPROVED** [review]

**Pull request merged**
